src/js/angular/controllers.js

```diff
@@ -951,6 +952,45 @@ function mainCtrl($scope, $menuItems, $jwtAuth, $http, toastr, $location, $repos
 
         return $filter('date')(time, ("'" + $translate.instant('timestamp.on') + "' yyyy-MM-dd '" + $translate.instant('timestamp.at') + "' HH:mm"));
     };
+
+    const updateAutocompleteStatus = () => {
+        if ($repositories.isActiveRepoFedXType() || !$licenseService.isLicenseValid()) {
+            WorkbenchContextService.setAutocompleteEnabled(false);
+            LocalStorageAdapter.set(LSKeys.AUTOCOMPLETE_ENABLED, false);
+            return;
+        }
+        AutocompleteService.checkAutocompleteStatus()
+            .then((autocompleteEnabled) => {
+                WorkbenchContextService.setAutocompleteEnabled(autocompleteEnabled);
+                LocalStorageAdapter.set(LSKeys.AUTOCOMPLETE_ENABLED, autocompleteEnabled);
+            });
+    };
+
+    const updateRepositoryNamespace = (activeRepository) => {
+        if (activeRepository) {
+            RDF4JRepositoriesService.getNamespaces(activeRepository)
+                .then((namespaces) => WorkbenchContextService.setSelectedRepositoryNamespaces(namespaces));
+        } else {
+            WorkbenchContextService.setSelectedRepositoryNamespaces(new NamespacesListModel([]));
+        }
+    };
+
+    const onRepositoriesChanged = () => {
+        const activeRepository = $repositories.getActiveRepository();
```

Comment:
BTW, its kind of bad what we are introducing now. Currently the repositories state is held in the the repositories service which is ugly and poorly written, yes, but now we introduce an additional place where we store the selected repository and I somehow don't like this